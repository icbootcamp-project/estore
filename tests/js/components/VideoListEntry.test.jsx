import React from 'react';
import VideoListEntry from '../../../src/js/components/VideoListEntry';

//Below are related to enzyme
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('VideoListEntry Component', () => {
  let handleSelectedVideo, wrapper;
  const video = {
    snippet: {
      title: 'some title',
      description: 'some description',
      thumbnails: {
        default: {
          url: 'someurl'
        }
      }
    }
  };

  beforeEach(() => {
    handleSelectedVideo = jest.fn();
    wrapper = shallow(
      <VideoListEntry handleSelectedVideo={handleSelectedVideo} video={video} />
    );
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('runs handleSelectedVideo on click handler', () => {
    wrapper.find('.media-heading').simulate('click');
    expect(handleSelectedVideo.mock.calls[0]).toEqual([video]);
  });
});
