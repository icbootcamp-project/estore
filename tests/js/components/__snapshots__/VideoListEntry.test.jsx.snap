// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoListEntry Component renders 1`] = `
<div
  className="videoListEntry"
>
  <div
    className="media"
  >
    <div
      className="media-left media-middle"
    >
      <img
        alt=""
        className="media-object"
        src="someurl"
      />
    </div>
    <div
      className="media-body"
    >
      <button
        className="media-heading"
        onClick={[Function]}
      >
        some title
      </button>
      <p
        className="media-description"
      >
        some description
      </p>
    </div>
  </div>
</div>
`;
